src/slices/AccountSlice.ts

```diff
@@ -50,18 +50,24 @@ export const getBalances = createAsyncThunk(
     const poolBalance = await poolTokenContract.balanceOf(address);
 
     let fsohmBalance = BigNumber.from(0);
-    for (const fuseAddressKey of ["FUSE_6_SOHM", "FUSE_18_SOHM", "FUSE_36_SOHM"]) {
-      if (addresses[networkID][fuseAddressKey]) {
-        const fsohmContract = new ethers.Contract(
-          addresses[networkID][fuseAddressKey] as string,
-          fuseProxy,
-          provider.getSigner(),
-        ) as FuseProxy;
-        // fsohmContract.signer;
-        const balanceOfUnderlying = await fsohmContract.callStatic.balanceOfUnderlying(address);
-        fsohmBalance = balanceOfUnderlying.add(fsohmBalance);
-      }
-    }
+
+    if (networkID === 1)
+      fsohmBalance = await Promise.all(
+        (["FUSE_6_SOHM", "FUSE_18_SOHM", "FUSE_36_SOHM"] as const).map(fuseAddressKey => {
+          const fsohmContract = new ethers.Contract(
+            addresses[networkID][fuseAddressKey] as string,
+            fuseProxy,
+            provider.getSigner(),
+          ) as FuseProxy;
+          // fsohmContract.signer;
+          return fsohmContract.callStatic.balanceOfUnderlying(address);
+        }),
+      ).then(underlyingBalances =>
```

Comment:
personally not the biggest fan of `.then`, but this is fine

Reply:
I'm not sure why I refactored this, maybe to use promise.all but what do you think of just wrapping the for loop with an `if` and call it a day